modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ObjectDefinitionExportImportTest.java

```diff
@@ -65,6 +82,129 @@ public void testExportImportObjectDefinition() throws Exception {
+				_defaultDraftObjectDefinitionJSONString
+			).put(
+				"active", true
+			).put(
+				"externalReferenceCode", externalReferenceCode
+			).put(
+				"name", name
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 0
+				).put(
+					"label", "approved"
+				).put(
+					"label_i18n", "Approved"
+				)
+			).toString();
+
+		testExportImportJSONString(
+			publishedObjectDefinitionJSONString,
+			publishedObjectDefinitionJSONString, externalReferenceCode, name);
+
+		// Cannot modify the status of published object definition
+
+		Class<?> clazz = getClass();
+
+		testFailedImportJSONString(
+			_jsonFactory.createJSONObject(
+				publishedObjectDefinitionJSONString
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 2
+				).put(
+					"label", "draft"
+				).put(
+					"label_i18n", "Draft"
+				)
+			).toString(),
+			StringUtil.read(
+				clazz.getResourceAsStream(
+					"dependencies/test-object-definition.status." +
+						"error-message.json")),
+			externalReferenceCode, name);
+
+		// Import bound object definitions one by one
+
+		String objectDefinition1JSONString = _jsonFactory.createJSONObject(
+			_defaultDraftObjectDefinitionJSONString
+		).put(
+			"externalReferenceCode", "TESTOBJECTDEFINITION1"
+		).put(
+			"name", "TestObjectDefinition1"
+		).put(
+			"objectRelationships",
+			JSONUtil.put(
+				_jsonFactory.createJSONObject(
+					_defaultObjectRelationshipJSONString
+				).put(
+					"edge", true
+				).put(
+					"name", "objectRelationship1"
+				).put(
+					"objectDefinitionExternalReferenceCode1",
+					"TESTOBJECTDEFINITION1"
+				).put(
+					"objectDefinitionExternalReferenceCode2",
+					"TESTOBJECTDEFINITION2"
+				).put(
+					"objectDefinitionName2", "TESTOBJECTDEFINITION2"
+				).put(
+					"type", "oneToMany"
+				))
+		).put(
+			"rootObjectDefinitionExternalReferenceCode", "TESTOBJECTDEFINITION1"
+		).toString();
+
+		testExportImportJSONString(
+			objectDefinition1JSONString, objectDefinition1JSONString,
+			"TESTOBJECTDEFINITION1", "TestObjectDefinition1");
+
+		Assert.assertNotNull(
+			_objectDefinitionResource.
+				getObjectDefinitionByExternalReferenceCode(
+					"TESTOBJECTDEFINITION2"));
+
+		String objectDefinition2JSONString = _jsonFactory.createJSONObject(
+			_defaultDraftObjectDefinitionJSONString
+		).put(
+			"externalReferenceCode", "TESTOBJECTDEFINITION2"
+		).put(
+			"name", "TestObjectDefinition2"
+		).put(
+			"rootObjectDefinitionExternalReferenceCode", "TESTOBJECTDEFINITION1"
+		).toString();
+
+		JSONObject expectedObjectDefinitionBoundJSONObject =
+			_jsonFactory.createJSONObject(objectDefinition2JSONString);
+
+		expectedObjectDefinitionBoundJSONObject.put(
+			"objectFields",
+			expectedObjectDefinitionBoundJSONObject.getJSONArray(
+				"objectFields"
+			).put(
+				_jsonFactory.createJSONObject(
+				).put(
+					"name", "r_objectRelationship1_c_testObjectDefinition1Id"
+				)
+			));
```

Comment:
Be careful with chaining methods. I could not find examples of chaining of:

- variable.getJSONArray().put()
- _jsonFactory.createJSONObject().put()

If you find any example, can you please show me?

Reply:
So, refactor this code to not chain methods :+1: